test/e2e/specs/site-editor/title.test.js

```diff
@@ -0,0 +1,109 @@
+/**
+ * WordPress dependencies
+ */
+const { test, expect } = require( '@wordpress/e2e-test-utils-playwright' );
+
+test.use( {
+	editorTitle: async ( { page, pageUtils, requestUtils }, use ) => {
+		await use( new EditorTitle( { page, pageUtils, requestUtils } ) );
+	},
+} );
+
+test.describe( 'Title', () => {
+	test.beforeAll( async ( { requestUtils } ) => {
+		await requestUtils.activateTheme( 'emptytheme' );
+		await Promise.all( [
+			requestUtils.deleteAllTemplates( 'wp_template' ),
+			requestUtils.deleteAllTemplates( 'wp_template_part' ),
+		] );
+	} );
+
+	test.afterEach( async ( { requestUtils } ) => {
+		await Promise.all( [
+			requestUtils.deleteAllTemplates( 'wp_template' ),
+			requestUtils.deleteAllTemplates( 'wp_template_part' ),
+		] );
+	} );
+
+	test.afterAll( async ( { requestUtils } ) => {
+		await requestUtils.activateTheme( 'twentytwentyone' );
+	} );
+
+	test( 'displays the selected template name in the title for the index template', async ( {
+		editorTitle,
+		pageUtils,
+	} ) => {
+		// Navigate to a template.
+		await pageUtils.visitSiteEditor( {
+			postId: 'emptytheme//index',
+			postType: 'wp_template',
+		} );
+
+		// Evaluate the document settings title.
+		const actual = await editorTitle.getEditorTitle();
+
+		await expect( actual ).toEqual( 'Editing template: Index' );
+	} );
+
+	test( 'displays the selected template name in the title for the header template', async ( {
+		editorTitle,
+		pageUtils,
+	} ) => {
+		// Navigate to a template part.
+		await pageUtils.visitSiteEditor( {
+			postId: 'emptytheme//header',
+			postType: 'wp_template_part',
+		} );
+
+		// Evaluate the document settings title.
+		const actual = await editorTitle.getEditorTitle();
+
+		await expect( actual ).toEqual( 'Editing template part: header' );
+	} );
+
+	test( "displays the selected template part's name in the secondary title when a template part is selected from List View", async ( {
+		editorTitle,
+		page,
+		pageUtils,
+	} ) => {
+		await pageUtils.visitSiteEditor( {
+			postId: 'emptytheme//index',
+			postType: 'wp_template',
+		} );
+
+		// Select the header template part via list view.
+		await page.click( 'role=button[name="List View"]' );
+		const listView = await page.locator(
+			'role=treegrid[name="Block navigation structure"]'
+		);
+		await listView.locator( 'a >> text="header"' ).click();
+		await page.click( 'role=button[name="Close List View Sidebar"]' );
+
+		// Evaluate the document settings secondary title.
+		const actual = await editorTitle.getEditorSecondaryTitle();
+
+		await expect( actual ).toEqual( 'header' );
+	} );
+} );
+
+class EditorTitle {
```

Comment:
If the class is supposed to be a POM, then the name should probably be `siteEditor` or something. Alternatively, since we have many tests for the site editor, it might be worth moving these common utils to `e2e-test-utils-playwright` and exporting it as a POM.

Reply:
I think I'll remove it and inline the code. After the change mentioned in https://github.com/WordPress/gutenberg/pull/40258#discussion_r848186763, the methods were both one liners.